### PR TITLE
Breed/cloud init configs

### PR DESCRIPTION
**Pull request by BRReed**

### CODE REVIEW

File: cloud-init-configs/ubuntu/destroy-jammy.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/bash/sh
+
+
+if !(virsh list --all | grep " jammy-cloud ") then
+  echo "jammy-cloud vm does not exist"
+else
+  echo "destroy, undefine and removing all storage for jammy-cloud vm"
+  virsh destroy jammy-cloud && virsh undefine jammy-cloud --remove-all-storage
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/destroy-jammy.sh:8]: Maybe rm user-data here too.

File: cloud-init-configs/ubuntu/destroy-jammy.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  echo "jammy-server-cloudimg-amd64.img does not exist"
+else
+  echo "removing jammy-server-cloudimg-amd64.img"
+  rm -f jammy-server-cloudimg-amd64.img 
+fi
+
+if !(find user-data) then
+  echo "user-data does not exist"
+else
+  echo "removing user-data"
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/destroy-jammy.sh:21]: ah here it is!

File: cloud-init-configs/ubuntu/spin-up-jammy.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+fi
+
+if (find jammy-server-cloudimg-amd64.img) then
+  echo "jammy-server-cloudimg-amd64.img exists"
+else
+  echo "Downloading latest jammy-server-cloudimg-amd64.img"
+  wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img
+fi
+
+cat << EOF > user-data
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/spin-up-jammy.sh]: Put this in the user data creating on line 8.

File: cloud-init-configs/ubuntu/spin-up-jammy.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+users:
+  - default
+  - name: brian
+    sudo: ["ALL-(ALL) NOPASSWD:ALL"]
+    shell: /bin/sh
+    ssh-authorized-keys:
+      - ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAINbliisKG/Rajx1joDff9Q6Kh9tLL+KD9Zqe4ZOsVttU b.r.reed@gmail.com
+EOF
+
+cloud-localds my-seed.img user-data meta-data
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/spin-up-jammy.sh]: How would you check if you have to do this or if it's already done?

File: cloud-init-configs/ubuntu/spin-up-jammy.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+qemu-img create -F qcow2 -b jammy-server-cloudimg-amd64.img -f qcow2 ./ub-jammy-vm.qcow2 50G
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/spin-up-jammy.sh]: How do you check for this if it's already done?

File: cloud-init-configs/ubuntu/spin-up-jammy.sh [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+virt-install --name jammy-cloud --memory 4096 --vcpus 4 --os-variant ubuntujammy --disk ./ub-jammy-vm.qcow2 --disk ./my-seed.img --import
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/spin-up-jammy.sh]: Same as above, put this in a conditional.

**Review by stephenlauck — CHANGES_REQUESTED** [review]

File: cloud-init-configs/ubuntu/spin-up-jammy.sh [hunk 7]
```diff
@@ -6,6 +6,17 @@ if (find user-data) then
 else
   echo "Creating user-data"
   touch user-data
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/spin-up-jammy.sh]: You don't need the `touch` here.

File: cloud-init-configs/ubuntu/spin-up-jammy.sh [hunk 8]
```diff
@@ -6,6 +6,17 @@ if (find user-data) then
...
+  cat << EOF > user-data
+#cloud-config
+
+users:
+  - default
+  - name: brian
+     sudo: ["ALL-(ALL) NOPASSWD:ALL"]
+    shell: /bin/sh
+    ssh-authorized-keys:
+      - ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAINbliisKG/Rajx1joDff9Q6Kh9tLL+KD9Zqe4ZOsVttU b.r.reed@gmail.com
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/spin-up-jammy.sh]: Is there a way to use a subshell here to get the key dynamically?

File: cloud-init-configs/ubuntu/spin-up-jammy.sh [hunk 9]
```diff
@@ -29,19 +40,23 @@ else
...
+  echo "ub-jammy-vm.qcow2 already exists"
+else
+  echo "creating ub-jammy-vm.qcow2"
+  qemu-img create -F qcow2 -b jammy-server-cloudimg-amd64.img -f qcow2 ./ub-jammy-vm.qcow2 50G
+fi
 
-qemu-img create -F qcow2 -b jammy-server-cloudimg-amd64.img -f qcow2 ./ub-jammy-vm.qcow2 50G
-virt-install --name jammy-cloud --memory 4096 --vcpus 4 --os-variant ubuntujammy --disk ./ub-jammy-vm.qcow2 --disk ./my-seed.img --import
\ No newline at end of file
+if (virsh domifaddr jammy-cloud) then
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/spin-up-jammy.sh]: You could also do `virsh list | grep running | grep VMNAME`.

Comment (BRReed) [reply cloud-init-configs/ubuntu/spin-up-jammy.sh]: changed -- 

Should I be using a variable for the vm name?

File: cloud-init-configs/ubuntu/spin-up-jammy.sh [hunk 10]
```diff
@@ -5,17 +5,17 @@ if (find user-data) then
   echo "user-data exists"
 else
   echo "Creating user-data"
-  touch user-data
-  cat << EOF > user-data
+  sshkey="$(cat ~/.ssh/id_ed25519.pub)"
```
Comment (stephenlauck) [inline cloud-init-configs/ubuntu/spin-up-jammy.sh]: Why are you setting a varialbe? Could you just put the subslhell on line 18?

Comment (BRReed) [reply cloud-init-configs/ubuntu/spin-up-jammy.sh]: I must have mistyped when I originally tried putting the subshell in the tee call. I didn't see an error so thought there was a collision between heredoc and shell commands. fixed now

**Review by stephenlauck — APPROVED** [review]

**Pull request merged**